soc/nordic/nrf54h/soc.c

```diff
@@ -122,6 +122,8 @@ bool z_arm_on_enter_cpu_idle(void)
 #ifdef CONFIG_LOG_FRONTEND_STMESP
 	log_frontend_stmesp_pre_sleep();
 #endif
+	nrf_lrcconf_poweron_force_set(NRF_LRCCONF010, NRF_LRCCONF_POWER_DOMAIN_0,
+			!IS_ENABLED(CONFIG_SOC_NRF54H20_CPURAD));
```

Comment:
This should be forced also for CPURAD.

I think we should have symmetry here. If restoring this register is done in `pm_state_exit_post_ops()`, can we configure it in `pm_state_enter_pre_ops()` or something like this?

Reply:
Also, this would conflict with clock_control configuring the same registers: https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/clock_control/clock_control_nrf2_common.c#L172-L173

Reply:
Just checked with PAC. In Radio CPU we don't need to `force_set()` to enter S2IDLE. So this condition is fine after all.

Reply:
> ~This should be forced also for CPURAD.~
> 
> I think we should have symmetry here. If restoring this register is done in `pm_state_exit_post_ops()`, can we configure it in `pm_state_enter_pre_ops()` or something like this?
The only "symmetric" function I've found is [pm_state_set](https://github.com/nrfconnect/sdk-zephyr/blob/main/include/zephyr/pm/pm.h#L139) which is defined for nRF54H20 [here](https://github.com/nrfconnect/sdk-zephyr/blob/main/soc/nordic/nrf54h/power.c#L131)

Reply:
Ok, then we should refactor k_cpu_idle() in such a way that it has symmetric hooks called from within its critical section. And we should aim to put this code in these hooks.

But until we get there I'm fine with currently used hooks. Hopefully, it unblocks progress with Bluetooth and other cases.

Reply:
This code makes `NRF_LRCCONF_POWER_DOMAIN_0` in Radio core to be always set to auto, no matter of its original state. That could collide with any other user of LCRCONF010 like clock control. Or am I wrong here?

Reply:
> Ok, then we should refactor k_cpu_idle() in such a way that it has symmetric hooks called from within its critical section. And we should aim to put this code in these hooks.
> 
> But until we get there I'm fine with currently used hooks. Hopefully, it unblocks progress with Bluetooth and other cases.

If only suspend-to-ram state is defined as power state and nothing else the system will still call `k_cpu_idle()` 
https://github.com/nrfconnect/sdk-zephyr/blob/main/kernel/idle.c#L85 and request poweron and never end up releasing it. This might still be acceptable to unblock the issue at hand but should still have a solution soon, otherwise we'll see different power consumptions based on how idle is used in combination with the pm subsystem.

Reply:
It will even happen when idle is defined as a pm state, as it might have an expected idle time below the threshold, such as in the case of our example overlay file here
https://github.com/nrfconnect/sdk-nrf/blob/84492d6d9264745050a018d01f4f333b285864c1/tests/benchmarks/multicore/idle/boards/nrf54h20dk_nrf54h20_cpuapp_s2ram.overlay#L12C39-L12C39
so a `k_msleep(10)` will not release the request